gdal/doc/source/development/rfc/rfc78_osgeo_utils.rst

```diff
@@ -0,0 +1,280 @@
+
+* https://pypi.org/project/GDAL/
+
+`osgeo_utils` as an independent wheel
+----------------------------------------
+
+This RFC suggests that in addition to the single `gdal` wheel, an initial version of a setup.py which builds an
+independent wheel will be available at `gdal/python/setup.py`.
+Using of this independent wheel will be evaluated later.
+
+The contents and layout of the new package
+---------------------------------------------
+
+GDAL is composed of:
+
+* A core written in C++.
+* SWIG bindings for a bunch of different languages.
+* Language specific code written on top of the SWIG bindings.
+  * Simple samples / scripts, partially documented without tests
+  * Modules that are documented, tested and well maintained independently of the GDAL core.
+
+Python code in this repo:
+
+* swig/python/osgeo - The swig bindings which generate the osgeo package.
+    As the SWIG bindings code tided to a specific GDAL version they will stay untouched tied to the
+    GDAL core in the `osgeo` package.
+
+* swig/python/osgeo/utils - Contains all the modules which have some functionality
+    on their own and have tests and documentation, As of GDAL=3.2 are reside in `osgeo.utils`.
+    The all would be moved to `osgeo_utils`.
+    As this subpackage was introduced only in GDAL 3.2 this RFC suggests not to be backward compatible
+    with the name `osgeo.utils` and in GDAL 3.3.0 release notes suggest this one character breaking change.
+
+* swig/python/osgeo/auxiliary - The `osgeo.auxiliary` subpackage which was introduced very recently
+    in Master as an auxiliary package for `osgeo.utils`.
+    It contains classes and functions that might not be useful by themselves and are used as
+    These shall be moved to `osgeo_utils.auxiliary`.
+
+* swig/python/samples - A bunch of sample scripts for the `osgeo` package.
+    A minimal documentation is available under the README.md file and no tests or backwards compatibility is checked.
+    This RFC suggests that they would be moved into `osgeo_utils.samples` so they can be found easily by users.
+
+* swig/python/scripts - deprecated scripts which are now just a trivial thin wrapper around each of the
+    osgeo.utils modules. These will still be here but will reference `osgeo_utils` instead of `osgeo.utils`.
+
+The layout of `osgeo_utils` would be:
+
+* `osgeo_utils` root - drop in replacement for `osgeo.utils`.
+
+* `osgeo_utils.auxiliary` - drop in replacement for `osgeo.auxiliary`.
+
+* `osgeo_utils.samples` - new subpackage for all the samples.
+
+* `osgeo_utils.base` - possible future subpackage for auxiliary modules that do not depend on the `osgeo` package.
+
+
+Folder structure change
+--------------------------
+
+* `gdal/swig/python/osgeo/utils` -> `gdal/python/src/osgeo_utils`
+
+* `gdal/swig/python/osgeo/auxiliary` -> `gdal/python/src/osgeo_utils/auxiliary`
+
+* `gdal/swig/python/scripts` -> `gdal/python/src/osgeo_utils/scripts`
+
+* `autotest/pyscripts` -> `gdal/python/tests`
```

Comment:
Some of the tests in autotest/pyscripts depend on autotest/pymod

Reply:
I see.
Do you think `pymod` needs to be a subpacakge of `osgeo_utils` ? or something else?

Reply:
but if autotest/pyscripts is moved in gdal/python/tests, it won't be tested anymore by running pytest from autotest  ? (I may have missed it from your implementation but the diff is too large to be easily reviewed)

Reply:
See [`autotest/pytest.ini`](https://github.com/OSGeo/gdal/pull/3259/files#diff-b4368b7fec3584d4c01ae8741e1a0df7a942e04713dd4f30b8e9c0d015ea9fb8)
`testpaths = ogr gcore gdrivers osr alg gnm utilities ../gdal/python/tests`